service/route_handlers.go

```diff
@@ -546,6 +574,59 @@ func HandleSamlAssertion(w http.ResponseWriter, r *http.Request, assertion *saml
 	if XForwardedProtoValue == "https" {
 		secure = true
 	}
+
+	/* token is created at this point. If the redirectBackBase is not same as the current URL's domain, then we cannot set the token as cookie
+	yet. So we will first send a 200 POST to /v1-auth/saml/tokenhtml, which will contain token cookie and redirectBackBase in HTML body,
+	and JS which will submit form on load to redirect URL. The handler for v1-auth/saml/tokenhtml will then set the token as cookie
+	*/
+
+	if redirectBackBaseValue != r.URL.Host {
```

Comment:
Does redirectBackBaseValue have the http:// portion in it? If so, will it always be different from r.URL.Host?